### PR TITLE
Add option to notify on fail and keep original notify

### PR DESCRIPTION
**Pull request by littleblack111**
#### Describe your PR, what does it fix/add?
By adding back original option to notify when successful and a notify-fail option for the need for original PR

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
i just added new option --notify-fail while keeping the --notify in place. So prev command still have the original functionality and we have the option for original PR

#### Is it ready for merging, or does it need work?
It removes the trivial of original PR

### CODE REVIEW

**Comment by KAGEYAM4** [comment]
@littleblack111  sorry for late reply, i somehow missed vaxry reply on hyprland repo.

So about this https://github.com/hyprwm/Hyprland/pull/8167#issuecomment-2423570036 , forget the first point. What are your suggestion on second (documentation change) and third points (code complexity) ?

File: hyprpm/src/main.cpp [hunk 1]
```diff
@@ -24,6 +24,7 @@ constexpr std::string_view HELP = R"#(┏ hyprpm, a Hyprland Plugin Manager
 ┣ Flags:
 ┃
 ┣ --notify       | -n    → Send a hyprland notification for important events (e.g. load fail)
+┣ --notify-fail  | -nn   → Send a hyprland notification for fail events
```
Comment (KAGEYAM4) [inline hyprpm/src/main.cpp]: Maybe being explicit about difference between -n and -nn

File: hyprpm/src/main.cpp [hunk 2]
```diff
@@ -52,6 +53,8 @@ int                        main(int argc, char** argv, char** envp) {
                 return 0;
             } else if (ARGS[i] == "--notify" || ARGS[i] == "-n") {
                 notify = true;
+            } else if (ARGS[i] == "--notify-fail" || ARGS[i] == "-nn") {
+                notifyFail = true;
```
Comment (KAGEYAM4) [inline hyprpm/src/main.cpp:57]:
```
notifyFail= notify = true;
```

File: hyprpm/src/main.cpp (hunk 1 above)
Comment (littleblack111) [reply hyprpm/src/main.cpp]: Sure, I'll add example for it too then

File: hyprpm/src/main.cpp [hunk 3]
```diff
@@ -113,7 +116,7 @@ int                        main(int argc, char** argv, char** envp) {
 
             if (ret2 != LOADSTATE_OK)
                 return 1;
-        } else if (notify)
+        } else if (notify || notifyFail)
```
Comment (KAGEYAM4) [inline hyprpm/src/main.cpp:119]:
```
else if (notify)
```

File: hyprpm/src/main.cpp [hunk 4]
```diff
@@ -146,7 +149,7 @@ int                        main(int argc, char** argv, char** envp) {
     } else if (command[0] == "reload") {
         auto ret = g_pPluginManager->ensurePluginsLoadState();
 
-        if (ret != LOADSTATE_OK && notify) {
+        if (ret != LOADSTATE_OK && (notify || notifyFail)) {
```
Comment (KAGEYAM4) [inline hyprpm/src/main.cpp:152]:
```
if (ret != LOADSTATE_OK &&  notify )
```

**Comment by littleblack111** [comment]
Yep, documentation is definitely needed and should be explicit about their differences.

And I think the third point is pretty straight forward, this ain't gonna be used as a lib or algorithm, just used a few times and not rly adding soo much to the time. So I'll make that change too

File: hyprpm/src/main.cpp (hunk 2 above)
Comment (KAGEYAM4) [inline hyprpm/src/main.cpp]:
```suggestion
                notifyFail = notify = true;
```

File: hyprpm/src/main.cpp (hunk 3 above)
Comment (KAGEYAM4) [inline hyprpm/src/main.cpp]:
```suggestion
        } else if (notify)
```

File: hyprpm/src/main.cpp (hunk 4 above)
Comment (KAGEYAM4) [inline hyprpm/src/main.cpp]:
```suggestion
        if (ret != LOADSTATE_OK && notify) {
```

**Pull request merged**